### PR TITLE
Networking stress tests moved out of Hosted pool

### LINKED ISSUE #34780: Out of disk space building on Docker NanoServer
stress-http and stress-ssl jobs are failing to build due to out of disk space, e.g.:
```
D:\a\1\s\artifacts\obj\coreclr\Linux.x64.Release\crossgen\src\utilcode\stdafx.utilcode_dac.cpp(1,1): fatal error C1085: Cannot write precompiled header file: 'D:/a/1/s/artifacts/obj/coreclr/Linux.x64.Release/crossgen/src/utilcode/Release/stdafx.utilcode_dac.pch': There is not enough space on the disk. [D:\a\1\s\artifacts\obj\coreclr\Linux.x64.Release\crossgen\src\utilcode\utilcode_dac.vcxproj]
  D:\a\1\s\artifacts\obj\coreclr\Linux.x64.Release\crossgen\src\md\runtime\Release\mdruntime_dac.lib : fatal error LNK1180: insufficient disk space to complete link [D:\a\1\s\artifacts\obj\coreclr\Linux.x64.Release\crossgen\src\md\runtime\mdruntime_dac.vcxproj]
##[error]BUILD: Error: cross-arch components build failed. Refer to the build log files for details.
```

https://dev.azure.com/dnceng/public/_build/results?buildId=592971&view=logs&j=2d2b3007-3c5c-5840-9bb0-2b1ea49925f3&t=abae1f68-3c73-5bff-491f-f2b908580ce6

https://dev.azure.com/dnceng/public/_build/results?buildId=592970&view=logs&j=2d2b3007-3c5c-5840-9bb0-2b1ea49925f3&t=abae1f68-3c73-5bff-491f-f2b908580ce6

@dotnet/runtime-infrastructure

### PR DESCRIPTION
**Pull request by alnikola**
Fixes #34780

### CODE REVIEW

**Review by safern — APPROVED** [review]

**Review by davidsh — APPROVED** [review]

**Review by ViktorHofer — APPROVED** [review]

**Comment by safern** [comment]
So the BYOC unix pools don't have powershell in them. I spoke with @alnikola and the most natural thing is to follow what we do for all our build scripts which is providing a Unix and a Windows build scripts to not require devs to install other dependencies in their machines and to have native support. I helped @alnikola by creating the Unix scripts. PTAL

cc: @eiriktsarpalis

**Comment by alnikola** [comment]
@MattGal It seems Windows machines don't have Docker at all and Linux ones don't have docker-compose. Is it possible to install these?

File: eng/docker/libraries-sdk.linux.Dockerfile
Comment (ViktorHofer) [inline eng/docker/libraries-sdk.linux.Dockerfile:11]: why are you changing this?

Comment (safern) [reply eng/docker/libraries-sdk.linux.Dockerfile:11]: The build were failing when building nuget packages in coreclr, I wanted to iterate quicker as CI was slow to test my changes. We can revert back... anyway, you know my take on building not necessary stuff on this builds.

Comment (ViktorHofer) [reply eng/docker/libraries-sdk.linux.Dockerfile:11]: FWIW,  https://github.com/dotnet/runtime/commit/11bf4b8f6e96a1b9deaef6470201d4e10f720b02#diff-f1921dfe4ae191cbfc80c608aa17dd4a would have broken this pipeline if we have had encoded the individual subsets here instead of just using the parent one `clr`. I'm fairly certain it wouldn't have been caught in the PR and we would have fixed it reactively. Just something that underlines my take on this.

Comment (safern) [reply eng/docker/libraries-sdk.linux.Dockerfile:11]: Yeah that was a breaking change that no one was aware of... it should've been done through the batched rollouts and now that subset is being removed... I understand the point, I'm just trying to help fix these builds without investing much time. Once the build is green, we can revert this change and investigate why building the coreclr packages fail on  these builds.

File: eng/docker/libraries-sdk.linux.Dockerfile
Comment (eiriktsarpalis) [inline eng/docker/libraries-sdk.linux.Dockerfile:8]: On line 1:
```
s/library artifacts/clr and library artifacts/
```

Would insert as an actual suggestion but github won't let me 🤷‍♂️

Comment (eiriktsarpalis) [inline eng/docker/libraries-sdk.linux.Dockerfile:8]: On line 2: the current tag for the target sdk image is `3.0.100` which is probably a bit too old. Unless the build argument is being overridden by the build script this should probably be updated?

Comment (alnikola) [reply eng/docker/libraries-sdk.linux.Dockerfile:8]: Fixed

Comment (alnikola) [reply eng/docker/libraries-sdk.linux.Dockerfile:8]: I probably agree that we need to update the base image version, but I think it should be done separately since this PR only deals with agent pool.

**Comment by MattGal** [comment]
> @MattGal It seems Windows machines don't have Docker at all and Linux ones don't have docker-compose. Is it possible to install these?

Filed https://github.com/dotnet/core-eng/issues/9641; I don't expect this to be a problem, we'll discuss it shortly.  If it's urgent reach out to @ilyas1974 , @tkapin, and myself and we'll chat.

**Comment by alnikola** [comment]
> Filed [dotnet/core-eng#9641](https://github.com/dotnet/core-eng/issues/9641); I don't expect this to be a problem, we'll discuss it shortly. If it's urgent reach out to @ilyas1974 , @tkapin, and myself and we'll chat.

I think we can wait until dotnet/core-eng#9641 is completed.

**Comment by alnikola** [comment]
@MattGal @ilyas1974 Could you please tell me if there is some progress on dotnet/core-eng#9641 (since it blocks my progress on this issue)?

**Comment by MattGal** [comment]
> @MattGal @ilyas1974 Could you please tell me if there is some progress on [dotnet/core-eng#9641](https://github.com/dotnet/core-eng/issues/9641) (since it blocks my progress on this issue)?

Not that I'm aware of, feel free to  ping the issue itself too.

**Comment by BruceForstall** [comment]
Is there any progress on this issue?

**Comment by MattGal** [comment]
> Is there any progress on this issue?

Yes!  The [linked PR](https://dev.azure.com/dnceng/internal/_git/dotnet-helix-machines/pullrequest/7626?_a=files)'s changes rolled out yesterday, so newly created machines in BuildPool.Server.Amd64.VS2019.* will have docker available, as well as BuildPool.Ubuntu.1604.Amd64.* and Ubuntu.1604.Amd64.* now having docker compose on them with their Docker.  

Please try it out and if you see issues with the buildpool.* machines (should only be an issue on buildpool.ubuntu.1604.amd64.open which needs to be cycled through to the new image) let us know.

**Comment by safern** [comment]
Thanks @MattGal -- I'll try to run the build again.

**Comment by alnikola** [comment]
@MattGal I see that Windows agents still don't have Docker installed, thus stress tests are failing there. Is there any plan to install Docker and docker-compose there as well?

**Comment by MattGal** [comment]
> @MattGal I see that Windows agents still don't have Docker installed, thus stress tests are failing there. Is there any plan to install Docker and docker-compose there as well?

That's odd as this work should have happened, taking a look.

File: eng/pipelines/libraries/stress/http.yml
Comment (MattGal) [inline eng/pipelines/libraries/stress/http.yml:56]: Build pool where docker was added is `BuildPool.Server.Amd64.VS2019`; should have same workloads as Windows.10 version

File: eng/pipelines/libraries/stress/ssl.yml
Comment (MattGal) [inline eng/pipelines/libraries/stress/ssl.yml:57]: Same as above, please try BuildPool.Server.Amd64.VS2019

**Comment by MattGal** [comment]
@alnikola added some comments, in our experience Windows docker is more stable for sustained use on server over client, so the BuildPool.Server.Amd64.VS2019.* queues were used.

**Comment by BruceForstall** [comment]
@alnikola Are you still working on this? The test runs continue to fail in the CI every day, and continue to be investigated by our vendor team.

**Comment by safern** [comment]
@MattGal it seems like the server pools don’t have docker-compose and these tests need it.

**Comment by alnikola** [comment]
@BruceForstall I'm tracking this, but we are currently blocked by missing docker-compose on BuildPool.Server.Amd64.VS2019.Open agents.
cc: @MattGal

File: eng/pipelines/libraries/stress/http.yml
Comment (alnikola) [reply eng/pipelines/libraries/stress/http.yml]: @MattGal  It's [still failing](https://dev.azure.com/dnceng/public/_build/results?buildId=667588&view=logs&j=2d2b3007-3c5c-5840-9bb0-2b1ea49925f3&t=ac0b0b0f-051f-52f7-8fb3-a7e384b0dde9&l=12) due to a missing docker-compose. Could you please share any ETA for fixing this? It's getting urgent for us since I noticed an instability on Linux tests and we need to have it merged into master and tracked regularly.

**Comment by MattGal** [comment]
> @BruceForstall I'm tracking this, but we are currently blocked by missing docker-compose on BuildPool.Server.Amd64.VS2019.Open agents.
> cc: @MattGal

Sorry about that.  I reactivated https://github.com/dotnet/core-eng/issues/9641 , @lukas-lansky FYI.    I think we assumed docker compose came with the windows install (it did not)

**Comment by alnikola** [comment]
@MattGal Windows tests successfully passed, thanks!
However, [on Linux they are still failing](https://dev.azure.com/dnceng/public/_build/results?buildId=681235&view=logs&j=0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb&t=6c56d998-158d-5232-283c-5140104799fc&l=14) due to missing docker-compose.

**Comment by alnikola** [comment]
@MattGal Please, disregard the above message. The SSL stress fail was caused by our mistake (missing file). Let me run HTTP stress tests to see if they work properly, then I will fix SSL project.

**Comment by alnikola** [comment]
@MattGal @lukas-lansky  Linux builds are getting cancelled due to exceeding 120 minutes quota. It happens [at step 6](https://dev.azure.com/dnceng/public/_build/results?buildId=682772&view=logs&j=0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb&t=6c56d998-158d-5232-283c-5140104799fc&l=28) where in contrast to Windows they are building libraries inside container which takes time. In the past they were doing it faster and fitted in the quota, but now it is apparently taking longer.
Could it be that VMs in the new pools are slower than the ones we used before? Or, can we increase time quota?

**Comment by safern** [comment]
@alnikola dumb question... is this step different in the linux vs windows nano tests? In Windows it is taking 46s to execute.

**Comment by MattGal** [comment]
> @MattGal @lukas-lansky Linux builds are getting cancelled due to exceeding 120 minutes quota. It happens [at step 6](https://dev.azure.com/dnceng/public/_build/results?buildId=682772&view=logs&j=0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb&t=6c56d998-158d-5232-283c-5140104799fc&l=28) where in contrast to Windows they are building libraries inside container which takes time. In the past they were doing it faster and fitted in the quota, but now it is apparently taking longer.
> Could it be that VMs in the new pools are slower than the ones we used before? Or, can we increase time quota?

No actually, a hosted Ubuntu machine is a Standard_DS2_v2  (2 core) and our build agents are D4_v3s (newer spec CPU and 4 cores), so I'd expect a slight improvement in performance over hosted. (from [this article](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml))

**Comment by alnikola** [comment]
@safern Yes, there is difference. Linux tests are building libraries inside the container. It was believed as a better way to set up test environment, but since it fails now with timeout I think we need to use the same approach as on Windows where we just copy artifacts.

**Comment by lukas-lansky** [comment]
@alnikola This particular 2 hour timeout is [under your control](https://github.com/dotnet/runtime/pull/35011/files#diff-53dfbdf75e79a04b57ab654be70b8cfcR28) I believe. You can try to tweak it to see whether this really is a performance problem.

**Comment by alnikola** [comment]
@safern @lukas-lansky I misunderstood the cause of failure. It doesn't hang on building libraries, because they were successfully built on the previous step "Build CLR and Libraries". It hangs on building the stress application itself which is quite unexplainable because it's a standard .Net Core application being built in a Docker container. Most likely, timeout has nothing to do with that, something else is failing out there.
I will probably try updating the base image version to a more recent and see if it helps.

**Comment by alnikola** [comment]
@MattGal @lukas-lansky Could you please check that Docker on Linux is healthy and has enough free space? It might be that libraries and SDK build consumed too much disk space on the agent (it consumes several GB), so Docker became unhealthy.

**Comment by MattGal** [comment]
> @MattGal @lukas-lansky Could you please check that Docker on Linux is healthy and has enough free space? It might be that libraries and SDK build consumed too much disk space on the agent (it consumes several GB), so Docker became unhealthy.

The term "Docker on linux" is somewhat vague but I'll look at your build and give any suggestions I can make.

**Comment by alnikola** [comment]
I basically meant if there is reasonably enough free disk space is available on the agent where we run Docker on Linux.

**Comment by MattGal** [comment]
> I basically meant if there is reasonably enough free disk space is available on the agent where we run Docker on Linux.

Yes there's a dedicated 512 GB disk only for Docker and Helix to use.  It looks like when your linux leg works, it's because that build hits an auth issue and never starts, and when it fails, that build starts and hangs.  I am still taking a look to see if I can glean anything else but it's very much not likely to be disk space.

**Comment by MattGal** [comment]
@alnikola I don't see any evidence of disk space issues here;  literally hundreds of jobs were able to run and succeed after this job, and there's tons of disk space.  It looks like there are several flaky points of automation; just about all the jobs where this "succeeds" the thing that builds the project actually fails like this:

```
Step 1/10 : ARG SDK_BASE_IMAGE=mcr.microsoft.com/dotnet/core/sdk:3.0.100-buster
Step 2/10 : FROM $SDK_BASE_IMAGE
Service 'server' failed to build: pull access denied for dotnet-sdk-libraries-current, repository does not exist or may require 'docker login': denied: requested access to the resource is denied

Finishing: Build HttpStress
```
In general the pipeline itself seems to have several ongoing issues that need to be cleaned up.

**Comment by alnikola** [comment]
> Step 1/10 : ARG SDK_BASE_IMAGE=mcr.microsoft.com/dotnet/core/sdk:3.0.100-buster
Step 2/10 : FROM $SDK_BASE_IMAGE
Service 'server' failed to build: pull access denied for dotnet-sdk-libraries-current, repository does not exist or may require 'docker login': denied: requested access to the resource is denied

@MattGal Could you please clarify where you see this error? [In the latest build](https://dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_apis/build/builds/682772/logs/16), step 1 to 5 completed successfully.

```
2020-06-11T15:53:29.5138847Z Building server
2020-06-11T15:53:29.6573548Z Step 1/10 : ARG SDK_BASE_IMAGE=mcr.microsoft.com/dotnet/core/sdk:3.0.100-buster
2020-06-11T15:53:29.6574635Z Step 2/10 : FROM $SDK_BASE_IMAGE
2020-06-11T15:53:29.6575742Z  ---> 75e4e0e61b92
2020-06-11T15:53:29.6576393Z Step 3/10 : WORKDIR /app
2020-06-11T15:53:29.8207510Z  ---> Running in f64c6083aeea
2020-06-11T15:53:30.7403308Z Removing intermediate container f64c6083aeea
2020-06-11T15:53:30.7405731Z  ---> f87e9eedc207
2020-06-11T15:53:30.7409085Z Step 4/10 : COPY . .
2020-06-11T15:53:31.6691038Z  ---> b45bf511bdbc
2020-06-11T15:53:31.6692029Z Step 5/10 : ARG CONFIGURATION=Release
2020-06-11T15:53:31.8097383Z  ---> Running in 00d19eeea372
2020-06-11T15:53:32.6870655Z Removing intermediate container 00d19eeea372
2020-06-11T15:53:32.6871921Z  ---> c6c0c3ae1c9e
2020-06-11T15:53:32.6872657Z Step 6/10 : RUN dotnet build -c $CONFIGURATION
2020-06-11T15:53:32.8247751Z  ---> Running in 86ffc7e8e56b
2020-06-11T15:53:34.1456754Z Microsoft (R) Build Engine version 16.3.0+0f4c62fea for .NET Core
2020-06-11T15:53:34.1458028Z Copyright (C) Microsoft Corporation. All rights reserved.
2020-06-11T15:53:34.1458444Z 
2020-06-11T15:53:36.5975673Z   Restore completed in 1.2 sec for /app/HttpStress.csproj.
2020-06-11T17:17:26.6297642Z ##[error]The operation was canceled.
2020-06-11T17:17:26.6323137Z ##[section]Finishing: Build HttpStress
```

**Comment by alnikola** [comment]
Ah, now I see that you got it from the master builds, but as far as I understand it's a side-effect of **low disk space** on an agent in Hosted pool and it's the very reason we are moving tests off that pool by this PR. 

As I mentioned above, step 2 successfully completes here on the new queue `BuildPool.Ubuntu.1604.Amd64.Open`. But, it now hangs at step 6.

Could you please check that Docker runtime on these new agents can correctly process at least some simple .Docker file? Basically, there is nothing unusual in our stress test .Docker file, so we need to confirm that Docker itself is properly configured.

**Comment by alnikola** [comment]
I'm also moving the libraries build out of the Linux containers as well to reduce amount of code executed inside Docker.

**Comment by MattGal** [comment]
> Ah, now I see that you got it from the master builds, but as far as I understand it's a side-effect of **low disk space** on an agent in Hosted pool and it's the very reason we are moving tests off that pool by this PR.
> 
> As I mentioned above, step 2 successfully completes here on the new queue `BuildPool.Ubuntu.1604.Amd64.Open`. But, it now hangs at step 6.
> 
> Could you please check that Docker runtime on these new agents can correctly process at least some simple .Docker file? Basically, there is nothing unusual in our stress test .Docker file, so we need to confirm that Docker itself is properly configured.

Are we doing DockerFile-s or docker-compose.ymls?  I can confirm that I verified using a `buildpool.ubuntu.1604.amd64.open` I could run `docker-compose up` of a hello world compose file [Log](https://helixri107v0xd6tv7v8isja.blob.core.windows.net/results-f8ebd014be6a4c9f8b/docker-compose-hello-world-0/console.44f4d5a8.log)

```
version: '2'
services:
  hello_world:
    image: ubuntu
    command: [/bin/echo, 'Hello world']
```

As for the regular Docker engine on these machines, pretty much every build uses a docker image so I'd be surprised if it had trouble.  If you have specific concerns about your scenario I'd like some more information about what you've tried for debugging and what happened;  the environment in the build is likely to be slightly different than your local box and without an assertion (something like "this feature is broken") it's very hard to help.

**Comment by alnikola** [comment]
@MattGal Previously, it was **docker** (not docker-compose) failure on `build clr+libs` command. However, I have now moved libraries' build out of the container on Linux (the same way as it's on Windows) and updated SDK image version. It will reduce coupling to Docker. Let's see how it goes.

**Comment by alnikola** [comment]
P.S. The issue was that Docker stopped sending any output after the following line ([full log](https://dev.azure.com/dnceng/9ee6d478-d288-47f7-aacc-f6e6d082ae6d/_apis/build/builds/682772/logs/16)), so I don't see what we can debug here. It seems as if we lost connection to the container.

```2020-06-11T15:53:32.8247751Z  ---> Running in 86ffc7e8e56b
2020-06-11T15:53:34.1456754Z Microsoft (R) Build Engine version 16.3.0+0f4c62fea for .NET Core
2020-06-11T15:53:34.1458028Z Copyright (C) Microsoft Corporation. All rights reserved.```

**Comment by alnikola** [comment]
@MattGal @lukas-lansky  In stress-http, the CoreCLR build on Linux **host** machine is failing and it is probably due to an outdated CMake version since it [prints a warning ](https://dev.azure.com/dnceng/public/_build/results?buildId=689543&view=logs&j=0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb&t=1451f5f3-0108-5a08-5b92-e984b2a85bbd&l=1041)right before the first error. Could you please update CMake there?

```
Found cmake v3.5.1 in PATH. Please install v3.14.5 or newer
```

**Comment by MattGal** [comment]
> @MattGal @lukas-lansky In stress-http, the CoreCLR build on Linux **host** machine is failing and it is probably due to an outdated CMake version since it [prints a warning ](https://dev.azure.com/dnceng/public/_build/results?buildId=689543&view=logs&j=0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb&t=1451f5f3-0108-5a08-5b92-e984b2a85bbd&l=1041)right before the first error. Could you please update CMake there?
> 
> ```
> Found cmake v3.5.1 in PATH. Please install v3.14.5 or newer
> ```

This is installed via the package manager on Ubuntu 16.04 and I believe represents the highest version available on the default feeds.  That is to say, having higher on Ubuntu 16.04 would require custom configuration.  Users tend to use docker images for builds here so it hasn't previously mattered.  Are you certain that this is why your **networking** stress test has a **network** connectivity issue. ?

**Comment by MattGal** [comment]
@alnikola I have experimentally verified using a fresh Ubuntu 16.04 VM that the latest CMake it can get from the official feeds is the version you're observing (`3.5.1-1ubuntu3`). We work hard to not have build environments that our users can only replicate by having to download and build their own custom versions of dependencies, add non-official package sources, or similar things.  

So, unless policy changes we won't be updating the cmake on the buildpool.ubuntu.1604 machines.  However, you should absolutely be able to build inside your own docker containers (as noted the main use case for folks on these build machines).  If you need more help debugging the building inside docker, let me know and I'll try to help.

**Comment by safern** [comment]
@alnikola  I think running the build inside a docker container is your best bet to control these sort of dependencies.

**Comment by alnikola** [comment]
I see, since the runtime build step is not failing by itself, so I can revert that change. But, could you please advise on how to fix the issue with the container stopping responding while building runtime? It hangs on a plain `dotnet build` command after printing 3 lines to the output as I showed above.

**Comment by MattGal** [comment]
> I see, since the runtime build step is not failing by itself, so I can revert that change. But, could you please advise on how to fix the issue with the container stopping responding while building runtime? It hangs on a plain `dotnet build` command after printing 3 lines to the output as I showed above.

The build is at a point where every attempt at this in the history chart for master fails 100% with `Service 'server' failed to build: pull access denied for dotnet-sdk-libraries-current, repository does not exist or may require 'docker login': denied: requested access to the resource is denied`

There's nothing unusual, or broken about docker on these build machines or we'd have users telling us, every day. You should be able to reproduce the build hang locally inside the same docker image, then add logging in the scripts being run til you understand where it's hanging, then guesswork from there ( it's linux docker, not the most user friendly place to debug stuff)

One other idea I had is that starting the docker container as priviliged might help...  especially if you try this locally on a Windows box and it doesn't reproduce as that concept isn't quite the same when hosting linux containers on Windows.

**Comment by safern** [comment]
Thanks @MattGal for the help here. I completely agree with you. 

@alnikola I'll try to repro this locally and see if I see anything obvious.

**Comment by alnikola** [comment]
@safern @MattGal I found the cause of current failures in master and opened the PR #38027
 to fix it. However, it won't solve "out of disk" problem, so we still have to move stress test off the Hosted pool.

**Comment by alnikola** [comment]
>  I found the cause of current failures in master and opened the PR #38027
to fix it.

As the matter of fact, that PR can fix only Linux tests. but on NanoServer they fail with actual HttpClient errors detected by stress test application. I see a lot of TaskCanceledExceptions thrown on `client` container while making calls to `server`, so it might be an infra issue caused by VM perf degradation. I will check if there are same errors in this PR running on a different queue.

**Comment by alnikola** [comment]
@MattGal I ran commands from http.yml building Linux tests locally on my Windows machine and confirmed that step 6 completes successfully (see below). Since I updated the base SDK image version to .NET 5.0, it now uses `Build Engine version 16.7.0` whereas previously it was `Build Engine version 16.3`, so it could be causing that hang. Let's wait and see if build succeeds on CI (SDK image version update was pushed yesterday)

```
Step 6/10 : RUN dotnet build -c $CONFIGURATION
 ---> Running in 935dd7d2147a
Microsoft (R) Build Engine version 16.7.0-preview-20310-07+ee1c9fd0c for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored /app/HttpStress.csproj (in 3.24 sec).
  You are using a preview version of .NET. See: https://aka.ms/dotnet-core-preview
  HttpStress -> /app/bin/Release/netcoreapp3.0/HttpStress.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:06.41
```

**Comment by alnikola** [comment]
@MattGal I fixed container builds on both of Windows and Linux for HttpStress and only Windows build for SslStress. On Linux build cannot execute `SslStress/run-docker-compose.sh` and [fails with `Permission denied`](https://dev.azure.com/dnceng/public/_build/results?buildId=691901&view=logs&j=50a69d0a-7972-5459-cdae-135ee6ebe312&t=ab661eaa-1aab-5546-d3f4-1113a638dfed&l=14). This command is [defined in ssl.yml](https://github.com/dotnet/runtime/blob/a130d74eddfa7f126b5d9e2b44dae607df4ef55c/eng/pipelines/libraries/stress/ssl.yml#L44), so it seems to be happening on the **host** OS. Could you please check why test runner process doesn't have required permissions?

This is the last issue preventing me from completing this PR. HttpStress tests are currently failing due to actual errors in the scenario and our team will check and prioritize them accordingly, but I have to complete this PR first.

**Comment by MattGal** [comment]
> @MattGal I fixed container builds on both of Windows and Linux for HttpStress and only Windows build for SslStress. On Linux SslStress container build cannot execute `SslStress/run-docker-compose.sh` and [fails with `Permission denied`](https://dev.azure.com/dnceng/public/_build/results?buildId=691901&view=logs&j=50a69d0a-7972-5459-cdae-135ee6ebe312&t=ab661eaa-1aab-5546-d3f4-1113a638dfed&l=14). This command is [defined in ssl.yml](https://github.com/dotnet/runtime/blob/a130d74eddfa7f126b5d9e2b44dae607df4ef55c/eng/pipelines/libraries/stress/ssl.yml#L44), so it seems to be happening on the **host** OS. Could you please check why test runner process doesn't have required permissions?
> 
> This is the last issue preventing me from completing this PR. HttpStress tests are currently failing due to actual errors in the scenario and our team will check and prioritize them accordingly, but I have to complete this PR first.

I'll take a look and see if I have anything useful to note;  do note the running user on these build agents has some options to unblock itself (will ping you on Teams about specifics)

**Comment by safern** [comment]
>  On Linux SslStress container build cannot execute SslStress/run-docker-compose.sh and fails with Permission denied. This command is defined in ssl.yml.

@alnikola we need to fix the permissions of the added script (chmod +x). I will push that now.

I will close the PR and trigger the pipelines manually from a branch to not waste resources (runtime, runtime-live-build, runtime-perf build, etc).

**Comment by safern** [comment]
Test build with chmod +x in the .sh file: https://dev.azure.com/dnceng/public/_build/results?buildId=692516&view=results

**Comment by safern** [comment]
@alnikola since I closed the PR it seems like I can't push to your branch; however it seems like the Linux ssl build is now fixed, just the windows one failing. This is the commit that fixes it so that you can cherry pick it: f3466adc2f8331e856f018cb7e53cab032b265d5

**Comment by alnikola** [comment]
@safern I repeated `chmod x` command on my branch and reopened this PR. Additionally, I'd like to clarify that HttpStress Windows build is failing due to some actual issues detected by the stress test. We will investigate it once this PR is merged.

**Comment by alnikola** [comment]
@MattGal @safern Strangely, but after I merged this into master Windows container builds started failing on `docker-compose` with a NativeCommandError without reporting any further details on both of [runtime-libraries http-stress](https://dev.azure.com/dnceng/public/_build/results?buildId=693943&view=logs&j=2d2b3007-3c5c-5840-9bb0-2b1ea49925f3&t=ac0b0b0f-051f-52f7-8fb3-a7e384b0dde9&l=15) and [runtime-libraries ssl-stress](https://dev.azure.com/dnceng/public/_build/results?buildId=693943&view=logs&j=2d2b3007-3c5c-5840-9bb0-2b1ea49925f3&t=ac0b0b0f-051f-52f7-8fb3-a7e384b0dde9&l=15). 
Could you please check what is going on? Could it be that `docker-compose` is not properly configured on some agents?

For the reference, these are pipeline runs I did on this PR right before merging. As you can see, `Build HttpStress` and `Build SslStress` completed successfully.
- [Last http-stress run in this PR](https://dev.azure.com/dnceng/public/_build/results?buildId=693632&view=logs&j=2d2b3007-3c5c-5840-9bb0-2b1ea49925f3&t=ac0b0b0f-051f-52f7-8fb3-a7e384b0dde9)
- [Last ssl-stress run in this PR](https://dev.azure.com/dnceng/public/_build/results?buildId=693633&view=logs&j=2d2b3007-3c5c-5840-9bb0-2b1ea49925f3&t=ac0b0b0f-051f-52f7-8fb3-a7e384b0dde9)

**Comment by MattGal** [comment]
> @MattGal @safern Strangely, but after I merged this into master Windows container builds started failing on `docker-compose` with a NativeCommandError without reporting any further details on both of [runtime-libraries http-stress](https://dev.azure.com/dnceng/public/_build/results?buildId=693943&view=logs&j=2d2b3007-3c5c-5840-9bb0-2b1ea49925f3&t=ac0b0b0f-051f-52f7-8fb3-a7e384b0dde9&l=15) and [runtime-libraries ssl-stress](https://dev.azure.com/dnceng/public/_build/results?buildId=693943&view=logs&j=2d2b3007-3c5c-5840-9bb0-2b1ea49925f3&t=ac0b0b0f-051f-52f7-8fb3-a7e384b0dde9&l=15).
> Could you please check what is going on? Could it be that `docker-compose` is not properly configured on some agents?
> 
> For the reference, these are pipeline runs I did on this PR right before merging. As you can see, `Build HttpStress` and `Build SslStress` completed successfully.
> 
> * [Last http-stress run in this PR](https://dev.azure.com/dnceng/public/_build/results?buildId=693632&view=logs&j=2d2b3007-3c5c-5840-9bb0-2b1ea49925f3&t=ac0b0b0f-051f-52f7-8fb3-a7e384b0dde9)
> * [Last ssl-stress run in this PR](https://dev.azure.com/dnceng/public/_build/results?buildId=693633&view=logs&j=2d2b3007-3c5c-5840-9bb0-2b1ea49925f3&t=ac0b0b0f-051f-52f7-8fb3-a7e384b0dde9)

I'm looking, but maybe we can hold off declaring "some machines are not properly configured" til this has worked a single time in master after your commit :)? All we know is it was tried once and failed.

All this error is telling us is that the thing you invoked wrote to standard error.  We get nothing else, and with the way it's being invoked we're not even getting its std out.

Example local ISE docker compose std err:
```
docker-compose : --build-arg is only supported when services are specified for API version < 1.25. Please use a Compose file version > 2.2 or specify which services to build.
At D:\scratch\compose\someshit.ps1:10 char:1
+ docker-compose --file "$COMPOSE_FILE" build $BUILD_ARGS.Split()
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (--build-arg is ...vices to build.:String) [], RemoteException
    + FullyQualifiedErrorId : NativeCommandError
```
That top line may be missed by console host.  Since you log none of the args being passed to the command, so I'd recommend adding logging to what the actual command you're running and all other values we can't see here; a few Write-Outputs can really explain things.

My first guess is that PRs have a different value for BUILD_CONFIGURATION than post-CI

**Pull request merged**
